### ISSUE TITLE
Update the app within Fox2Code itself!

### ISSUE THREAD

**Issue by CodElixer** [Fox2Code/FoxMagiskModuleManager#170]
Updating the latest version of Fox2Code app within the app itself would be reliable than referencing the GitHub page even the older app is already installed on the device!

I think you should consider for updating the app within itself for future Fox2Code versions.

**Comment by Fox2Code** [comment]
I'm thinking about it, I think there is a way to do it nicely, but it will take time.

**Comment by Fox2Code** [comment]
Yes, and it should have been opened in a different issue, I will be working on fixing this later.

**Comment by HiFiiDev** [comment]
@Fox2Code AOSPMods has a nice way they integrated updates for their module/app, I wonder if you could do the same? 

Alternatively (and this just came to my mind) You could probably even just have a telegram channel for app updates and one to talk to the users, so you would just simply post on telegram that a new update is available. 

Or, maybe something like this[This](https://www.make.com/en/integrations/github/telegram?fromImt=1)  that could just simply do all the work.  Compile, pull a changelog, and post it on telegram

If these ideas are no good hopefully they will at least get some sort of ball rolling for you if you still plan to implement this

**Comment by aioros57** [comment]
Hello, a little complicated to explain and long.
Easier to look at the comments on XDA on this problem I am not the only one to have this problem with the fox manager application from magisk
[]https://forum.xda-developers.com/attachments/screenshot_20220826-151354_foxs-magisk-module-manager-png.5696767/(url)
You see the concern that has when there is an update of the application this message in red, following the reboot of the device no more root and magisk just the application install in magisk magisk 25.2 no install and zygisk no more root .

**Comment by HuskyDG** [comment]
@aioros57 please open new issue for this

**Comment by Fox2Code** [comment]
I'm doing some clean-up on this issue because there is too much unrelated things here.

Peoples that do not open new issues for things unrelated here will just get their report thrown away.

Make new issue here: https://github.com/Fox2Code/FoxMagiskModuleManager/issues/new/choose

**Comment by TipsyTheCat** [comment]
I suggest making this feature a toggle in the settings, for those who use an f-droid client.
Or, even better, have this feature only in the GitHub releases.
